src/LBPair.sol

```diff
@@ -900,24 +923,39 @@ contract LBPair is LBToken, ReentrancyGuardUpgradeable, Clone, ILBPair {
      * @dev Returns the address of the token X
      * @return The address of the token X
      */
-    function _tokenX() internal pure returns (IERC20) {
-        return IERC20(_getArgAddress(0));
+    function _tokenX() internal view returns (IERC20) {
+        address tokenX_;
+        bytes32 slot = 0x3441ab29b24daf7a3fd59500b0e08396ec08ec96f5cc2d0362924cdd45cfec31; //keccak256(abi.encode(uint256(keccak256("sovrynlbdex.pair.storage.TokenX")) - 1));
+        assembly {
+            tokenX_ := sload(slot)
+        }
+
+        return IERC20(tokenX_);
     }
 
     /**
      * @dev Returns the address of the token Y
      * @return The address of the token Y
      */
-    function _tokenY() internal pure returns (IERC20) {
-        return IERC20(_getArgAddress(20));
+    function _tokenY() internal view returns (IERC20) {
+        address tokenY_;
+        bytes32 slot = 0x7e1935766b7c49e7482a018a5ee52ca183a2ddfcb6810787916934079aa58264; // keccak256(abi.encode(uint256(keccak256("sovrynlbdex.pair.storage.TokenY")) - 1));
```

Comment:
Ditto.

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/ab08ddafa24bc76b0d41843f366ac1d6fcf439ff)